internal/exec/util/file.go

```diff
@@ -152,7 +153,8 @@ func (u Util) WriteLink(s types.Link) error {
 
 func (u Util) SetPermissions(mode *int, node types.Node) error {
 	if mode != nil {
-		mode := os.FileMode(*mode)
+		m := cutil.IntToPtr(int(toFileMode(mode)))
+		mode := os.FileMode(*m)
```

Comment:
We should be able to drop the casting/pointering/depointering here.  It should be fine for `toFileMode` to take an `int`.